src/litdata/processing/data_processor.py

```diff
@@ -606,10 +609,16 @@ def _loop(self) -> None:
                     return
                 continue
 
-            assert isinstance(combined_data, tuple), f"Invalid data received from queue {combined_data=}."
-            assert len(combined_data) == 3, f"Invalid data received from queue {combined_data=}."
+            if self.using_queue_optimize:
+                # using random index number, just to satisfy the default combined_data structure.
+                index = -5  # not using -1, as items will be of length 0, so last_index (0-1 = -1) will become true.
+                item = combined_data
+                paths = None
+            else:
+                assert isinstance(combined_data, tuple), f"Invalid data received from queue {combined_data=}."
```

Comment:
These `assert`s are user-facing, since they rely on what the user emits from the queue in the processor function, right?
In such case we should favor raising exceptions, and use assert for checking invariants developer-facing.